test_no_accidental_warnings fails with packaging >= 20.5

numba 0.53 on openSUSE build service for Tumbleweed:

* Python 3.8.8
* packaging 20.8
* setuptools 44.1.1

```
[  745s] ======================================================================
[  745s] FAIL: test_no_accidental_warnings (numba.tests.test_import.TestNumbaImport)
[  745s] ----------------------------------------------------------------------
[  745s] Traceback (most recent call last):
[  745s]   File "/usr/lib64/python3.8/site-packages/numba/tests/test_import.py", line 62, in test_no_accidental_warnings
[  745s]     self.run_in_subproc(code, flags)
[  745s]   File "/usr/lib64/python3.8/site-packages/numba/tests/test_import.py", line 22, in run_in_subproc
[  745s]     raise AssertionError(msg % (popen.returncode, err.decode()))
[  745s] AssertionError: process failed with code 1: stderr follows
[  745s] Traceback (most recent call last):
[  745s]   File "/usr/lib/python3.8/site-packages/packaging/version.py", line 57, in parse
[  745s]     return Version(version)
[  745s]   File "/usr/lib/python3.8/site-packages/packaging/version.py", line 298, in __init__
[  745s]     raise InvalidVersion("Invalid version: '{0}'".format(version))
[  745s] packaging.version.InvalidVersion: Invalid version: 'numpy'
[  745s] 
[  745s] During handling of the above exception, another exception occurred:
[  745s] 
[  745s] Traceback (most recent call last):
[  745s]   File "<string>", line 1, in <module>
[  745s]   File "/usr/lib64/python3.8/site-packages/numba/__init__.py", line 19, in <module>
[  745s]     from numba.core import config
[  745s]   File "/usr/lib64/python3.8/site-packages/numba/core/config.py", line 16, in <module>
[  745s]     import llvmlite.binding as ll
[  745s]   File "/usr/lib64/python3.8/site-packages/llvmlite/binding/__init__.py", line 4, in <module>
[  745s]     from .dylib import *
[  745s]   File "/usr/lib64/python3.8/site-packages/llvmlite/binding/dylib.py", line 3, in <module>
[  745s]     from llvmlite.binding import ffi
[  745s]   File "/usr/lib64/python3.8/site-packages/llvmlite/binding/ffi.py", line 175, in <module>
[  745s]     from pkg_resources import resource_filename
[  745s]   File "/usr/lib/python3.8/site-packages/pkg_resources/__init__.py", line 3252, in <module>
[  745s]     def _initialize_master_working_set():
[  745s]   File "/usr/lib/python3.8/site-packages/pkg_resources/__init__.py", line 3235, in _call_aside
[  745s]     f(*args, **kwargs)
[  745s]   File "/usr/lib/python3.8/site-packages/pkg_resources/__init__.py", line 3264, in _initialize_master_working_set
[  745s]     working_set = WorkingSet._build_master()
[  745s]   File "/usr/lib/python3.8/site-packages/pkg_resources/__init__.py", line 574, in _build_master
[  745s]     ws = cls()
[  745s]   File "/usr/lib/python3.8/site-packages/pkg_resources/__init__.py", line 567, in __init__
[  745s]     self.add_entry(entry)
[  745s]   File "/usr/lib/python3.8/site-packages/pkg_resources/__init__.py", line 623, in add_entry
[  745s]     for dist in find_distributions(entry, True):
[  745s]   File "/usr/lib/python3.8/site-packages/pkg_resources/__init__.py", line 2061, in find_on_path
[  745s]     path_item_entries = _by_version_descending(filtered)
[  745s]   File "/usr/lib/python3.8/site-packages/pkg_resources/__init__.py", line 2034, in _by_version_descending
[  745s]     return sorted(names, key=_by_version, reverse=True)
[  745s]   File "/usr/lib/python3.8/site-packages/pkg_resources/__init__.py", line 2032, in _by_version
[  745s]     return [packaging.version.parse(part) for part in parts]
[  745s]   File "/usr/lib/python3.8/site-packages/pkg_resources/__init__.py", line 2032, in <listcomp>
[  745s]     return [packaging.version.parse(part) for part in parts]
[  745s]   File "/usr/lib/python3.8/site-packages/packaging/version.py", line 59, in parse
[  745s]     return LegacyVersion(version)
[  745s]   File "/usr/lib/python3.8/site-packages/packaging/version.py", line 127, in __init__
[  745s]     warnings.warn(
[  745s] DeprecationWarning: Creating a LegacyVersion has been deprecated and will be removed in the next major release

```

This is another fallout of https://github.com/pypa/setuptools/issues/2466

--- a/numba/tests/test_import.py
+++ b/numba/tests/test_import.py
@@ -60,7 +60,7 @@ class TestNumbaImport(TestCase):
         code = "import numba"
         # See: https://github.com/numba/numba/issues/6831
         # bug in setuptools/packaging causing a deprecation warning
-        flags = ["-Werror", "-Wignore::DeprecationWarning:packaging:"]
+        flags = ["-Werror", "-Wignore::DeprecationWarning:packaging.version:"]
         self.run_in_subproc(code, flags)
 
     def test_import_star(self):

--- a/numba/tests/test_import.py
+++ b/numba/tests/test_import.py
@@ -58,7 +58,9 @@ class TestNumbaImport(TestCase):
         # checks that importing Numba isn't accidentally triggering warnings due
         # to e.g. deprecated use of import locations from Python's stdlib
         code = "import numba"
-        flags = ["-Werror",]
+        # See: https://github.com/numba/numba/issues/6831
+        # bug in setuptools/packaging causing a deprecation warning
+        flags = ["-Werror", "-Wignore::DeprecationWarning:packaging:"]
         self.run_in_subproc(code, flags)
 
     def test_import_star(self):

